frontend/server/src/SecurityTools.php

```diff
@@ -249,12 +249,68 @@ public static function getGitserverAuthorizationToken(
      *
      * @param array<string, string> $claims
      * @param string $subject
+     * @param string $issuer
      * @return string The Bearer authorization token.
      */
     public static function getCourseCloneAuthorizationToken(
         array $claims,
-        string $subject
+        string $subject,
+        string $issuer
     ): string {
+        $secretKey = self::getCourseCloneSecretKey();
+        $token = (new \ParagonIE\Paseto\Builder())
+            ->setKey($secretKey)
+            ->setVersion(new \ParagonIE\Paseto\Protocol\Version2())
+            ->setPurpose(\ParagonIE\Paseto\Purpose::local())
+            ->setExpiration(
+                (new \DateTime())
+                    ->setTimestamp(\OmegaUp\Time::get())
+                    ->add(new \DateInterval('P7D'))
+            )
+            ->setIssuer($issuer)
+            ->setSubject($subject)
+            ->setClaims($claims);
+        return $token->toString();
+    }
+
+    /**
+     * @return array<string, string>
+     */
+    public static function getDecodedCloneCourseToken(
+        string $token,
+        string $courseAlias
+    ): array {
+        $parser = \ParagonIE\Paseto\Parser::getLocal(
+            self::getCourseCloneSecretKey(),
+            \ParagonIE\Paseto\ProtocolCollection::v2()
+        )
+            ->addRule(
+                new \ParagonIE\Paseto\Rules\ValidAt(
+                    (new \DateTime())->setTimestamp(\OmegaUp\Time::get())
+                )
+            )
+            ->addRule(new \OmegaUp\ClaimRule('course', $courseAlias))
+            ->addRule(new \OmegaUp\ClaimRule('permissions', 'clone'));
+        try {
+            $parsedToken = $parser->parse($token, /*$skipValidation=*/true);
+        } catch (\ParagonIE\Paseto\Exception\PasetoException $e) {
+            throw new \OmegaUp\Exceptions\TokenDecodeException();
+        }
```

Comment:
```suggestion
        $parsedToken = $parser->parse($token, /*$skipValidation=*/true);
```

aquí se vale dejar que esa excepción se propague porque en el código que llama a esto no _necesita_ referirse a esta excepción por nombre. con `} catch (Exception $e) {` basta. onda

```php
try {
  ....
} catch (\OmegaUp\Exceptions\TokenValidateException $e) {
   // aquí se puede usar $e->claims para agregarlo al log.
   ....
} catch (Exception $e) {
   // aquí no se puede usar los claims porque todo falló miserablemente.
   ...
}
```

Reply:
Listo